h/static/scripts/annotator/plugin/pdf-metadata.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+/**
+ * This PDFMetadata service provides implementations for extracting fields
+ * of a pdf document from the input PDFViewerApplication object.
+ *
+ * The fields extracted are:
+ * #1. urn - a string which uniquely identifies a pdf document.
+ * #2. title - a string representing the title of the pdf document.
+ * #3. link - an array of two objects, the urn and the url strings of the pdf.
+ * #4. documentFingerprint - the string representing the pdf fingerprint.
+ */
+function PDFMetadata(pdfViewerApplication) {
+  this._pdfViewerApplication = pdfViewerApplication;
+  this._documentLoaded = new Promise(function (resolve) {
+    var finish;
+    finish = function () {
+      window.removeEventListener('documentload', finish);
+      resolve();
+    };
+
+    if (this._pdfViewerApplication.documentFingerprint) {
+      resolve();
+    }
+
+    window.addEventListener('documentload', finish);
+  }.bind(this));
```

Comment:
Using `Function.bind` can get a bit messy if you have nested scopes. It's more common in our code base to use the closure pattern:

```javascript
var self = this;

function () {
    self.foo();
}
```